comms/src/blazingdb/manager/Context.cc

```diff
@@ -15,7 +15,15 @@ Context::Context(const uint32_t token,
       masterNode_{masterNode},
       logicalPlan_{logicalPlan},
       query_step{0},
-      query_substep{0} {}
+      query_substep{0},
+      kernel_id_{0} {}
+
+std::shared_ptr<Context> Context::clone() {
```

Comment:
Why is htere a clone here. Wouldn't properly implementing a copy constructor be the best way to handle this?